### PR TITLE
Revive the `av stack reparent` command

### PR DESCRIPTION
**Pull request by draftcode**

### CODE REVIEW

File: cmd/av/stack_reparent.go [hunk 1]
```diff
@@ -1,24 +1,281 @@
...
+		return vm, tea.Quit
+	case *stackReparentState:
+		vm.state = msg
+		if stackReparentFlags.Skip || stackReparentFlags.Continue || stackReparentFlags.Abort {
+			if stackReparentFlags.Abort {
+				vm.abortedBranch = vm.state.Seq.CurrentSyncRef
+			}
+			return vm, vm.runSeqWithContinuationFlags
+		}
+		return vm, vm.runSeq
```
Comment (jainankit) [inline cmd/av/stack_reparent.go:109]: skip and continue have the same behavior?

File: internal/utils/stackutils/stackutils.go [hunk 2]
```diff
@@ -89,15 +89,38 @@ func BuildStackTreeAllBranches(tx meta.ReadTx, currentBranch string, sortCurrent
...
+		if err != nil {
+			// Ignore branches that are not adopted to av.
+			continue
+		}
+		for _, name := range stack {
+			branches[name] = true
+		}
+	}
+	var names []string
+	for name := range branches {
```
Comment (jainankit) [inline internal/utils/stackutils/stackutils.go:115]: Is this just to turn valid branches into a list?

File: cmd/av/stack_reparent.go [hunk 3]
```diff
@@ -1,24 +1,281 @@
...
+}
+
+func (vm stackReparentViewModel) View() string {
+	sb := strings.Builder{}
+	if vm.state != nil && vm.state.Seq != nil {
+		sb.WriteString("Reparenting " + vm.state.InitialBranch + " onto " + vm.state.NewParentBranch + "...\n")
+		if vm.state.Seq.CurrentSyncRef != "" {
+			sb.WriteString("Restacking " + vm.state.Seq.CurrentSyncRef.Short() + "...\n")
+		} else if vm.abortedBranch != "" {
+			sb.WriteString("Restack aborted\n")
```
Comment (jainankit) [inline cmd/av/stack_reparent.go:154]: if the action is aborted would we want to return here early?

File: cmd/av/stack_reparent.go [hunk 4]
```diff
@@ -1,24 +1,281 @@
...
+		} else {
+			sb.WriteString("Restack done\n")
+		}
+		syncedBranches := map[plumbing.ReferenceName]bool{}
+		pendingBranches := map[plumbing.ReferenceName]bool{}
+		seenCurrent := false
+		for _, op := range vm.state.Seq.Operations {
+			if op.Name == vm.state.Seq.CurrentSyncRef || op.Name == vm.abortedBranch {
+				seenCurrent = true
```
Comment (jainankit) [inline cmd/av/stack_reparent.go:163]: not sure I'm following the logic here. Some comment could be helpful.

File: cmd/av/stack_reparent.go (hunk 1 above)
Comment (draftcode) [reply cmd/av/stack_reparent.go:109]: They are different. https://git-scm.com/docs/git-rebase#_mode_options

File: cmd/av/stack_reparent.go (hunk 3 above)
Comment (draftcode) [reply cmd/av/stack_reparent.go:154]: No. We want to show the tree still.

File: internal/utils/stackutils/stackutils.go (hunk 2 above)
Comment (draftcode) [reply internal/utils/stackutils/stackutils.go:115]: Yes.

File: cmd/av/stack_reparent.go (hunk 4 above)
Comment (draftcode) [reply cmd/av/stack_reparent.go:166]: Done.

**Review by jainankit — APPROVED** [review]

**Pull request closed without merging**